### ISSUE TITLE
checkin-test.py no longer functions

### ISSUE THREAD

**Issue by csiefer2** [trilinos/Trilinos#1335]
@bartlettroscoe 
@bmpersc 

The python checkin script no longer functions.  It requires yaml_cpp, but cannot find it even if the associated sems module is loaded.

**Comment by bartlettroscoe** [comment]
@csiefer2, There must be a problem with your local SEMS env.  I am in SNL this week.  I will stop by your office.

**Comment by bartlettroscoe** [comment]
@csiefer2 and I just met to look at this.  So it looks like his machine is set up so that running checkin-test-sems.sh from his csh shell causes the module load command to fail inside of Trilinos/cmake/load_sems_dev_env.sh.  I tried to reproduce this on my machine by running:

```
cd CHECKIN/
$ csh
$ ./checkin-test-sems.sh
```

and it loaded the modules correctly.  I wonder if there is some trick to setting up the sems modules so that the csh use case as set up on Chris's machine will work correctly?   Given that that checkin-test-sems.sh opens a new bash shell, csh should have nothing to do with correctly running this script.

In any case, when we switched to 'bash' on Chris's machine and run the checkin-test-sems.sh script, it ran fine.

@csiefer2, therefore, if you can just open a separate terminal on your machine and then run `bash` in that terminal, then you should be able to run `checkin-test-sems.sh --do-all --push` for all of your pushes without issue.

FYI: Just note that ROL and Teko currently have failing tests (see #1304) so please take those into account.  You can disable these tests by adding the following to your invocation of the checkin-test-sems.sh script:

```
---cmake-extra-options="-DROL_example_burgers-control_example_04_MPI_1_DISABLE=ON -DROL_example_PDE-OPT_topo-opt_elasticity_example_01_MPI_4_DISABLE=ON -DROL_test_sol_solSROMGenerator_MPI_1_DISABLE=ON -DTeko_DiagonallyScaledPreconditioner_MPI_1_DISABLE=ON"
```

as mentioned at the bottom of:

* https://github.com/trilinos/Trilinos/wiki/Policies-|-Safe-Checkin-Testing

Sorry we are still experiencing this instability but as more people use the checkin-test-sems.sh script, the better this gets.

**Comment by bartlettroscoe** [comment]
FYI:

I added the note:

* https://github.com/trilinos/Trilinos/wiki/Policies-|-Safe-Checkin-Testing#csh_shell_issues

to mention csh issues and how to address them.  Directly loading the SEMS modules with `load_[ci_]sems_dev_env.sh` from a `(t)csh` shell is not currently supported.  However, I think we find a way to support maintainable `load_[ci_]sems_dev_env.csh` scripts if that is a major need. But there should be no need to support a `checkin-test-sems.csh` script since `checkin-test-sems.sh` opens its own shell when it runs.  Instead, we should address why module commands.  But note that others have seen with the module command, for example [here](https://github.com/trilinos/Trilinos/issues/482#issuecomment-289928854).

**Comment by mhoemmen** [comment]
@bartlettroscoe Would it make sense to add yaml-cpp to `Trilinos/packages/common/auxiliarySoftware`, assuming that its license allows this?  That could save users the trouble of needing yet another TPL.

**Comment by lxmota** [comment]
@mhoemmen On the issue of yaml-cpp, the license allows inclusion:

https://github.com/jbeder/yaml-cpp/blob/master/LICENSE

**Comment by mhoemmen** [comment]
@lxmota thanks! :-) 
@bartlettroscoe If we put yaml-cpp in `auxiliarySoftware`, would that preclude users from specifying their own build of yaml-cpp?

**Comment by ibaned** [comment]
Although I think this particular issue should be fixed by focusing on `csh` compatibility, I just wanted to comment that if `yaml-cpp` is put in `auxiliarySoftware`, it would be preferable to use the latest version from GitHub, which has several key advantages over the SEMS released version:
 - doesn't depend on Boost
 - doesn't reorder the parameters arbitrarily (keep the in-file order)
 - users get fewer compiler warnings (I contributed some of that)

**Comment by bartlettroscoe** [comment]
> If we put yaml-cpp in auxiliarySoftware, would that preclude users from specifying their own build of yaml-cpp?

Given the current setup of TriBITS, that would be hard.  But once we can get to the refactoring in TriBITSPub/TriBITS#63, that would be easy to support.  In fact, once that refactoring is done, I want default builds for all of the basic upstream TPLs like BLAS, LAPACK, etc.  The details on that would be determined after TriBITSPub/TriBITS#63 is complete (hopefully by the end of FY17).

**Comment by bartlettroscoe** [comment]
@csiefer2, I think we established that this is not a checkin-test.py or checkin-test-sems.sh issue.   And since you were able to push the commit:

```
commit ce1a5f6f5a599903dbe740380193f06a7cfebf46
Author:     Chris Siefert <csiefer@sandia.gov>
AuthorDate: Fri May 19 11:59:40 2017 -0600
Commit:     Chris Siefert <csiefer@sandia.gov>
CommitDate: Fri May 19 14:51:24 2017 -0600

    MueLu: Adding option to crash on "isolated" nodes in Phase 3 Aggregation.
    
    Basically, if you find a non-Dirichlet node w/ no on-rank neighbords during
    Phase-3, you throw and error if this option is set.
    
    This guy is very useful if you're running with dim(nullspace) != numPDEs
    (e.g. elasticity) because it will prevent you from accidently creating
    a singular mesh at coarse levels.
    
    Build/Test Cases Summary
    Enabled Packages: MueLu
    Disabled Packages: PyTrilinos,Claps,TriKota
    Enabled all Forward Packages
    0) MPI_RELEASE_DEBUG_SHARED_PT => passed: passed=420,notpassed=0 (30.48 min)
```

it shows that you were able to  successfully use the checkin-test-sems.sh script.

If we just require people to use 'bash' and run the checkin-test-sems.sh script, there should be no issue.  If there is a desire for direct `csh` shell support, then that needs to be a separate issue.

Can we close this Issue?  Since you are the customer and opened the issue, you should be the one to close it.  (It should be an offical Trilinos policy that only customers close issues that they create).  For now I will just put it in review to make it clear this is just waiting for the customer review.

**Comment by bartlettroscoe** [comment]
> Can we close this Issue? Since you are the customer and opened the issue, you should be the one to close it. (It should be an offical Trilinos policy that only customers close issues that they create). For now I will just put it in review to make it clear this is just waiting for the customer review.

No response for 12 days from customer so I am closing.  As I stated above, there was nothing wrong with the checkin-test.py or checkin-test-sems.sh scripts (other than a `csh` issue that is now documented at https://github.com/trilinos/Trilinos/wiki/Policies-%7C-Safe-Checkin-Testing#csh_shell_issues).
